Date: Tue, 6 Aug 2024 16:40:46 +0100
From: Sean Young <sean@mess.org>
Subject: Re: [PATCH 6.10 288/809] media: dvb-usb: Fix unexpected infinite
 loop in dvb_usb_read_remote_control()
Message-ID: <ZrJD_gHZCsphqT-U@gofer.mess.org>

I don't think this drivers uses the bulk endpoint, and it is missing the
corresponding out bulk endpoint.

Please could you test the patch below please - that would be very helpful in
narrowing down this issue.

Thank you!

Sean

From adda4b83a6b800d4b35bbd12b9a198fb6a085f58 Mon Sep 17 00:00:00 2001
From: Sean Young <sean@mess.org>
Date: Tue, 6 Aug 2024 16:31:02 +0100
Subject: [PATCH] media: dw2102: TeVii DVB-S2 S660 does not have bulk endpoint

Since commit 2052138b7da5 ("media: dvb-usb: Fix unexpected infinite loop
in dvb_usb_read_remote_control()"), we check that the bulk endpoints exist.
This device does have them and then errors out during initialization.

Fixes: 2052138b7da5 ("media: dvb-usb: Fix unexpected infinite loop in dvb_usb_read_remote_control()")
Signed-off-by: Sean Young <sean@mess.org>
---
 drivers/media/usb/dvb-usb/dw2102.c | 1 -
 1 file changed, 1 deletion(-)

--- a/drivers/media/usb/dvb-usb/dw2102.c
+++ b/drivers/media/usb/dvb-usb/dw2102.c
@@ -2250,7 +2250,6 @@ static struct dvb_usb_device_properties
 		.rc_query = dw2102_rc_query,
 	},
 
-	.generic_bulk_ctrl_endpoint = 0x81,
 	.num_adapters = 1,
 	.download_firmware = dw2102_load_firmware,
 	.read_mac_address = s6x0_read_mac_address,
